core/trino-main/src/main/java/io/trino/sql/gen/ExpressionCompiler.java

```diff
@@ -131,12 +139,7 @@ public Supplier<PageProcessor> compilePageProcessor(Optional<RowExpression> filt
     private <T> Class<? extends T> compile(Optional<RowExpression> filter, List<RowExpression> projections, BodyCompiler bodyCompiler, Class<? extends T> superType)
     {
         // create filter and project page iterator class
-        try {
-            return compileProcessor(filter.orElse(constant(true, BOOLEAN)), projections, bodyCompiler, superType);
-        }
-        catch (CompilationException e) {
-            throw new TrinoException(COMPILER_ERROR, e.getCause());
```

Comment:
why removed?

Reply:
For going to the 2nd catch block in  `LocalExecutionPlanner.Visitor#visitScanFilterAndProject`. Updated a commit body. 
```java
            catch (TrinoException e) {
                throw e;
            }
            catch (RuntimeException e) {
                throw new TrinoException(
                        COMPILER_ERROR,
                        "Compiler failed. Possible reasons include: the query may have too many or too complex expressions, " +
                                "or the underlying tables may have too many columns",
                        e);
            }
```